docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -187,6 +187,31 @@ declare module '@mui/material/Button' {
 
 {{"demo": "GlobalThemeVariants.js"}}
 
+The variant's props property can be defined in a form of a callback too. This is useful if you want to apply some styles if you need to use a negation in the condition, for e.g. apply some styles if some property does not have some value.
```

Comment:
```suggestion
The variant `props` can also be defined as a callback. 
This is useful if you want to apply styles when using negation in the condition.
In other words, applying a different style if a particular property doesn't have a specific value.
```

Quick writing suggestion! A few thoughts:
- "variant's prop property" sounded a bit confusing & redundant 😅 hopefully this is clear?
- just "e.g." already means "for example", so there is no need for "for". But, in any case, maybe it's better to avoid latinisms entirely, as the reviewdog says 😬!